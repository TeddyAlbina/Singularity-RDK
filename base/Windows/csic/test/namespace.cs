using N;

namespace N {
using System;
using XX = System.Console;

class A {
	protected internal void f() { XX.WriteLine("A.f()"); }
	protected internal static void g() { System.Console.WriteLine("A.g()"); }
	public class X {}
	public static X x;
}
class B: A {
	new public int f = '3';
	public new static void g() { XX.WriteLine("B.g()"); A.x = new X(); }
}
}

class test {
	public static System.IO.TextWriter w = System.Console.Out;
	public static void Main() {
		B a = new B();
		w.WriteLine("a.f={0}", a.f);
		((A)a).f();
		A.g();
		B.g();
		w.WriteLine("A.x={0}", A.x);
	}
}
