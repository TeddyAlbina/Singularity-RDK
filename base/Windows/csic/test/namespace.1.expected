a.f=51
A.f()
A.g()
B.g()
A.x=N.A+X
